sycl/include/sycl/esimd/esimd_enum.hpp

```diff
@@ -0,0 +1,137 @@
+//==---------------- esimd_enum.hpp - DPC++ Explicit SIMD API   ------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// definitions used in Explicit SIMD APIs.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/defines.hpp>
+#include <cstdint> // for uint* types
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace intel {
+namespace gpu {
+
+// Typedefs
+typedef unsigned char uchar;
+typedef unsigned short ushort;
+typedef unsigned int uint;
+
+typedef unsigned char U8;
+typedef unsigned short U16;
+typedef unsigned int U32;
+
+#ifdef __SYCL_DEVICE_ONLY__
+// Mark a function being nodebug.
+#define ESIMD_NODEBUG __attribute__((nodebug))
+// Mark a "ESIMD global": accessible from all functions in current translation
+// unit, separate copy per subgroup (work-item), mapped to SPIRV private storage
+// class.
+#define ESIMD_PRIVATE __attribute__((opencl_private))
+// Bind a ESIMD global variable to a specific register.
+#define ESIMD_REGISTER(n) __attribute__((register(n)))
+#else
+// TODO ESIMD define what this means on Windows host
+#define ESIMD_NODEBUG
+// On host device ESIMD global is a thread local static var. This assumes that
+// each work-item is mapped to a separate OS thread on host device.
+#define ESIMD_PRIVATE thread_local
+#define ESIMD_REGISTER(n)
+#endif
+
+// Mark a function being noinline
+#define ESIMD_NOINLINE __attribute__((noinline))
```

Comment:
Are these not __SYCL_DEVICE_ONLY__ dependent?

Reply:
No, since they can be used in the code that can run on the HOST as well.